influxdb3_py_api/src/system_py.rs

```diff
@@ -899,6 +939,324 @@ fn process_response_part(
+    }
+
+    fn is_empty(&self) -> bool {
+        self.entries.is_empty()
+    }
+}
+
+#[pyclass]
+#[derive(Hash, Eq, PartialEq, Debug, Clone)]
+enum CacheId {
+    Global(),
+    GlobalTest(String),
+    Trigger {
+        database: String,
+        trigger_name: String,
+    },
+    TriggerTest(String),
+}
+
+impl CacheId {
+    fn default_expiration(&self) -> Option<Duration> {
+        match self {
+            Global() | Trigger { .. } => None,
+            // for tests, keep values around for 30 minutes.
+            GlobalTest(_) | TriggerTest(_) => Some(Duration::from_secs(30 * 60)),
+        }
+    }
+}
+pub fn drop_trigger_cache(database: String, trigger_name: String) -> bool {
+    let mut cache = GLOBAL_CACHE_STORE.lock();
+    let Some(cache) = cache.as_mut() else {
+        return false;
+    };
+    cache
+        .namespaces
+        .remove(&Trigger {
+            database,
+            trigger_name,
+        })
+        .is_some()
+}
+
+impl CacheStore {
+    fn new(time_provider: Arc<dyn TimeProvider>, cleanup_interval: Duration) -> Self {
+        let last_cleanup = time_provider.now();
+        Self {
+            namespaces: HashMap::new(),
+            time_provider,
+            cleanup_interval,
+            last_cleanup,
+        }
+    }
+
+    fn should_run_cleanup(&self) -> bool {
+        self.time_provider
+            .now()
+            .checked_duration_since(self.last_cleanup)
+            .unwrap_or_default()
+            >= self.cleanup_interval
+    }
+
+    fn cleanup(&mut self) {
+        if !self.should_run_cleanup() {
+            return;
+        }
+        for cache in self.namespaces.values_mut() {
+            cache.cleanup();
+        }
+
+        self.namespaces.retain(|_, cache| !cache.is_empty());
+    }
+
+    fn put(&mut self, cache_id: &CacheId, key: &str, value: Py<PyAny>, ttl: Option<f64>) {
+        let entry = CacheEntry::new(value, ttl, Arc::clone(&self.time_provider));
+        if let Some(cache) = self.namespaces.get_mut(cache_id) {
+            cache.insert(key.to_string(), entry);
+            return;
+        }
+        self.namespaces
+            .entry(cache_id.clone())
+            .or_insert_with(|| {
+                ExpiringCache::new(
+                    Arc::clone(&self.time_provider),
+                    cache_id.default_expiration(),
+                )
+            })
+            .insert(key.to_string(), entry);
+    }
+
+    fn get(&mut self, cache_id: &CacheId, py: Python<'_>, key: &str) -> Option<Py<PyAny>> {
+        let cache = self.namespaces.get_mut(cache_id)?;
+
+        cache.get(key, py)
+    }
+
+    fn delete(&mut self, cache_id: &CacheId, key: &str) -> bool {
+        if let Some(cache) = self.namespaces.get_mut(cache_id) {
+            cache.remove(key)
+        } else {
+            false
+        }
+    }
+}
+
+// Thread-local cache store to safely manage Python objects
+lazy_static::lazy_static! {
```

Comment:
Thought this was deprecated a while ago because it's now standard in Rust? https://github.com/rust-lang-nursery/lazy-static.rs/issues/214